source/includes/dev/_microservice.md

```diff
@@ -126,14 +218,81 @@ As explained above, although the call is ultimately to another microservice, the
+
+### Elasticsearch v5.5
+
+Use [Elasticsearch](https://www.elastic.co/) for search optimization or heterogeneous data storage with quick access. The current setup also includes [a 3rd party plugin for SQL support on ES 5](https://github.com/NLPchina/elasticsearch-sql).
+
+To see an example of how to use Elasticsearch on a real-world microservice, check out the [Document dataset adapter microservice](https://github.com/resource-watch/document-adapter/).
+
+### Redis v5.0
+
+[Redis](https://redis.io/) is an in-memory data storage tool, and can also be used as a [pub-sub](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) messaging tool.
+
+You can learn how to use Redis in your applications by looking at the code of the [Subscriptions microservice](https://github.com/gfw-api/gfw-subscription-api).
+
+### Neo4J v2.0
+
+[Neo4J](https://neo4j.com/) is a graph database used by [Graph microservice](https://github.com/resource-watch/graph-client/) to build complex associations between different RW API resources.
 
-### Logging
+### RabbitMQ v3.7
+
+[RabbitMQ](https://www.rabbitmq.com/) is a message broker service, which is particularly useful when handling long, asynchronous operations. You can see an example of its usage on the [Document microservice - Executor submodule](https://github.com/resource-watch/doc-executor) code base.
+
+### Cloud services
+
+Some microservices have data storage needs that are not covered by the applications described here (for example, file storage). In those scenarios, it's common to use cloud services (like AWS S3, for example), but do reach out to the broader RW API development team before implementing your solution.
+
+## HTTP caching
+
+The RW API has a system-wide HTTP cache that you may use to cache your requests, improving scalability and response times. This cache is based on [Fastly](https://www.fastly.com/), and you can browse its documentation if you are looking for a specific detail on its behavior. For most common use cases, you just need to keep in mind the following:
+
+- Only responses with codes 200, 203, 300 and 410 are cached, and the default cache TTL is 3 days.
+- GET responses for `/auth` endpoints are never cached.
+- GET responses for `/query` or `/fields` endpoints are cached for 2 days.
+- Your microservice can use the `cache` response header to tag a cache entry. This, in itself, has no functional impact. Example [here](https://github.com/resource-watch/dataset/blob/47ad8b9509b97803d7f484549908e72ecaa98467/app/src/routes/api/v1/dataset.router.js#L396).
+- Your microservice can use the `uncache` header to purge cache entries matching a given tag, as set in the previous point. Example [here](https://github.com/resource-watch/dataset/blob/47ad8b9509b97803d7f484549908e72ecaa98467/app/src/routes/api/v1/dataset.router.js#L462).
+
+## Logging
```

Comment:
I'd add, somewhere in this section, that you can (**and should!**) configure the error reporting level in most MSs using the `LOGGER_LEVEL` env variable.